### PR TITLE
ci: Reimplemented mergebot

### PR DESCRIPTION
**Pull request by shaeed-noordeen**
**What problem does this PR solve?**:

Added mergebot back to charts as mergebot can now function in public repos.

**Which issue(s) does this PR fix?**:

https://jira.d2iq.com/browse/D2IQ-90583

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**If the PR adds a version bump, does it add a breaking change in License**:

- [ ] No License Change (or NA).

### CODE REVIEW

**Comment by shaeed-noordeen** [comment]
Please remove old mergebot webhook

**Comment by cbuto** [comment]
@shaeed-noordeen looks like the pre-commit check is failing because your commit does not follow conventional commits

```
1: CT1 Title does not follow ConventionalCommits.org format 'type(optional-scope): description': "Create mergebot.yml"
```

if you reword your commit to something like `ci: create mergebot.yml` and force push it should be good

**Comment by gracedo** [comment]
I force pushed the commit message fix

**Review by gracedo — APPROVED** [review]

File: .github/workflows/mergebot.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        run: git config --global user.name d2iq-mergebot
+      - name: Set mergebot email
+        run: git config --global user.email ci-mergebot@d2iq.com
+      - name: Set gpg key id
+        run: git config --global user.signingKey ${{ secrets.MERGEBOT_GPG_KEY_ID }}
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v4
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
```
Comment (gracedo) [inline .github/workflows/mergebot.yml:28]: I don't think this secret is defined

File: .github/workflows/mergebot.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Run Mergebot
+        env:
+          LOG_STDOUT: "true"
+          EVENT_TYPE: ${{ github.event_name }}
+          RUN_ID: ${{ github.run_id }}
+          MERGEBOT_TC_USERNAME: ${{secrets.MERGEBOT_TC_USERNAME}}
+          MERGEBOT_TC_PASSWORD: ${{secrets.MERGEBOT_TC_PASSWORD}}
+          MERGEBOT_TC_TOKEN: ${{secrets.MERGEBOT_TC_TOKEN}}
+          JIRA_USERNAME: ${{secrets.JIRA_USERNAME}}
+          JIRA_PASSWORD: ${{secrets.JIRA_PASSWORD}}
```
Comment (gracedo) [inline .github/workflows/mergebot.yml:53]: I don't think these secrets are defined

**Comment by shaeed-noordeen** [comment]
@gracedo Those are org level secrets, so they're indeed there.

**Review by cbuto — APPROVED** [review]

**Comment by gracedo** [comment]
@shaeed-noordeen just checking, but do they need to be authorized? 
image

**Comment by shaeed-noordeen** [comment]
> @shaeed-noordeen just checking, but do they need to be authorized? 
> 
> image
> 
> 

That's weird, could you check the repository secrets. I told Kevin to add it there if it wasn't, so it should be there. Furthermore, mergebot would've failed to work if they weren't there, but it looks like it's worked on open, sync and label event and has no failures. So everything seems to be in order.

**Comment by gracedo** [comment]
@shaeed-noordeen We just have these secrets defined at the repo level
image

**Comment by shaeed-noordeen** [comment]
> @shaeed-noordeen We just have these secrets defined at the repo level image

I see, I'll sort this out asap. Thanks for checking!

**Comment by shaeed-noordeen** [comment]
> @shaeed-noordeen We just have these secrets defined at the repo level
> 
> image
> 
> 

Org secrets have now been added

**Comment by gracedo** [comment]
@shaeed-noordeen I think we're just missing `secrets.GPG_PASSPHRASE` now!

**Comment by shaeed-noordeen** [comment]
> @shaeed-noordeen I think we're just missing `secrets.GPG_PASSPHRASE` now!

Sorry for the late reply, I was out sick. That is already set as an org secret, here's a screenshot. 
Screenshot 2022-09-27 at 16 47 31

**Comment by gracedo** [comment]
@shaeed-noordeen These are the org level secrets that I see are authorized for this particular repo
image

**Pull request merged**